### PR TITLE
fault-proof: kzg point evaluation precompile

### PR DESCRIPTION
**Pull request by Inphi**
Specify oracle interaction for KZG point evaluation precompile results.

### CODE REVIEW

File: specs/experimental/fault-proof/index.md
Comment (tynes) [inline specs/experimental/fault-proof/index.md:163]: Why is this necessary? We stub out the execution of the precompile in the program with a lookup to the oracle?

Comment (Inphi) [reply specs/experimental/fault-proof/index.md:163]: That's exactly it. Executing the precompile in the program is terribly expensive for cannon. It's much cheaper to offload this to the oracle.
This involves changes to op-geth to facilitate this in the op-program (https://github.com/ethereum-optimism/op-geth/pull/242).

Comment (tynes) [reply specs/experimental/fault-proof/index.md:163]: Would love this kind of context to be included in the specs

Comment (Inphi) [reply specs/experimental/fault-proof/index.md:163]: Good suggestion, thanks. I've added a section detailing this.

File: specs/experimental/fault-proof/index.md
Comment (ajsutton) [inline specs/experimental/fault-proof/index.md]:
```suggestion
An EIP-4844 point-evaluation precompile result. It maps directly to the EIP-4844
```

?

File: specs/experimental/fault-proof/index.md
Comment (ajsutton) [inline specs/experimental/fault-proof/index.md]:
```suggestion
This approach ensures that the fault proof program can complete a state transition in a reasonable
```
?

**Review by ajsutton — APPROVED** [review]
LGTM. Couple of suggestions but not blockers.  Probably would be good to get @tynes approval before merging as well.

**Review by tynes — APPROVED** [review]
Looks good to me! Appreciate the extra details :)

**Pull request merged**
